src/clr-angular/utils/_mixins.clarity.scss

```diff
@@ -82,3 +84,19 @@ $clr-outline-spread: 2px;
     }
   }
 }
+
+@mixin css-var($style: null, $var: null, $fallback: '', $do-custom-prop: true) {
+  @if $style != null and $var != null {
+    /* #{$var} */
+    #{$style}: $fallback;
+    @if $do-custom-prop == true {
+      #{$style}: var(css-variablize-string($var), $fallback);
+    }
+  }
+}
```

Comment:
my intent was to prevent the function from having to be aware of anything outside of itself. some degree of purity there.

while it's a little annoying right now to have to pass in `$clr-use-custom-properties`, we are (as you say) avoiding any weird scoping issues, avoiding having to make sure global variables are available for mixins because of this function, and in 4.0 it all goes away. We could literally just remove the `$do-custom-prop` parameter from the declaration and the SASS compiler will tell us where we need to update the function calls.